internal/provider/helpers.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/gookit/color"
+)
+
+type tfOperation string
+
+const (
+	CreateOperation tfOperation = "CREATE"
+	ReadOperation   tfOperation = "READ"
+	UpdateOperation tfOperation = "UPDATE"
+	DeleteOperation tfOperation = "DELETE"
+)
+
+func formatSQLPreview(operation tfOperation, resourceName string, id string, commands []string) string {
+	var c color.Color
+	switch operation {
+	case CreateOperation:
+		c = color.HiGreen
+	case ReadOperation:
+		c = color.HiBlue
+	case UpdateOperation:
+		c = color.HiYellow
+	case DeleteOperation:
+		c = color.HiRed
+	}
+	var sb strings.Builder
+	sb.WriteString(c.Sprintf("\n[ %s %s %s ]", operation, resourceName, id))
+	for _, command := range commands {
+		sb.WriteString(c.Sprintf("\n  - %s", command))
+	}
+	sb.WriteString("\n")
+	return sb.String()
+}
+
+type sensitiveAttributes struct {
+	m map[string]bool
+}
+
+var (
+	sa   *sensitiveAttributes
+	lock = sync.Mutex{}
+)
+
+func isSensitive(s string) bool {
+	if sa == nil {
+		lock.Lock()
+		defer lock.Unlock()
+		if sa == nil {
+			sa = &sensitiveAttributes{
+				m: make(map[string]bool),
+			}
+			dir, err := os.UserHomeDir()
+			if err != nil {
+				return false
+			}
+			// sensitive path is ~/.snowflake/sensitive.
+			f := filepath.Join(dir, ".snowflake", "sensitive")
+			dat, err := os.ReadFile(f)
+			if err != nil {
+				return false
+			}
+			lines := strings.Split(string(dat), "\n")
+			r := regexp.MustCompile("(data[.])?snowflake_(.*)[.](.+)[.](.+)")
```

Comment:
is this regexp coming from some terraform doc? Let's test it and document it.

Reply:
this is a resource address. Supposed to be you have a `~/.snowflake/sensitive` file containing contents of:

```
snowflake_resource_monitor.*.credit_quota
snowflake_resource_monitor.*.frequency
```
And this would match the resource addresses included there